test/ci/vars.yml.sample

```diff
@@ -0,0 +1,38 @@
+---
+vm_prefix: "ci_test"
+
+type: aws
+aws_user: "ec2-user"
+python: "/usr/bin/python"
+aws_key: "libra"
+aws_region: "us-east-1"
+aws_cluster_id: "ci"
+# us-east-1d
+aws_subnet: "subnet-cf57c596"
+
+aws_ami_tags:
+  operating_system: "rhel"
+  image_stage: "base"
+  ready: "yes"
+
+aws_instances:
+- name: "{{ vm_prefix }}-master"
+  groups:
+    - masters
+    - etcd
+    - nodes
+  aws_flavor: t2.large
+  aws_security_group: public
+  node_group: "node-config-all-in-one"
```

Comment:
This could be better handled by group vars.

Possible inventory groups:
```yaml
[OSEv3:children]
etcd
nodes

[nodes:children]
masters
compute

[etcd:children]
masters

[masters]
master1.example.com

[master:vars]
openshift_node_group_name: node-config-master-infra

[compute]
node1.example.com

[compute:vars]
openshift_node_group_name: node-config-compute
```

Obviously, we can define vars sections in appropriate group_vars files and we can adjust groups for different scenarios.

Reply:
That'd complicate `vars.yml` - a list of groups to create and assign the instance to the group? In the end I'd still have to set per-instance `aws_ip` and `aws_id`, so why not put `openshift_node_group_name` there too?

Reply:
> That'd complicate vars.yml - a list of groups to create and assign the instance to the group? In the end I'd still have to set per-instance aws_ip and aws_id, so why not put openshift_node_group_name there too?

I'm not sure what you mean.  You don't need to 'create' the groups, you just make this the base template of the inventory file (since we're not using dynamic inventory).

Reply:
That'd limit us in groups choice. Instead of hardcoding the list of those in the template these groups are now being created automagically. So later on we could try scale up tests, by adding hosts in `new_nodes` group for instance.

Reply:
I'm not a fan of this implementation, I do think we need to update our group reference architecture to reflect what an actual user should do as far as group hierarchy.  I'm against using hostvars where we could use groupvars because the syntax is different between the two in ini files and users never seem to understand what hostvars actually are (you'll see inventories with same host in two groups, eg masters and nodes, and competing values set on each instance of the host).

In any case, we can revisit this later.